task_1/cluster_tf/main.tf

```diff
@@ -0,0 +1,76 @@
+module "eks" {
+  source       = "github.com/terraform-aws-modules/terraform-aws-eks?ref=v11.1.0"
+  cluster_name = var.cluster_name
+  subnets      = var.private_subnets
+  vpc_id       = var.vpc_id
+  tags         = merge({ "Name" = var.cluster_name }, var.tags)
+
+  node_groups_defaults = {
+    ami_type  = "AL2_x86_64"
+    disk_size = var.node_group_settings["disk_size"]
+  }
+
+  node_groups = {
```

Comment:
How would you need to change that, if for example, we would like to add a 50% spot instances to the pool of worker nodes in EKS?

Reply:
it can be done in two different ways:
- separate autoscaling group can be created solely for spot instances, it needs to be tagged appropriately so it is discovered by cluster autoscaler.
- create single autoscaling group which runs mixed on-demand and spot instances in 50/50.